batoms/batoms.py

```diff
@@ -435,6 +435,39 @@ def from_batoms(self, label):
         self._species = Bspecies(label, {}, self)
         self._attributes = Attributes(label=label, parent=self, obj_name=self.obj_name)
         self.selects = Selects(label, self)
+    
+    @classmethod
+    def from_dict(cls, data):
+        """Build batoms object from a dictionary.
+
+        Args:
+            data (dict): _description_
+
+
+        Returns:
+            Batoms: _description_
+
+        """
+        pass
+
+    def as_dict(self):
```

Comment:
looks good. maybe in the next version, we can allow as_dict to output information of any plugins

Reply:
Yes, I will implement this.

Reply:
I made the following changes:
- avoid explicitly loading plugins inside Batoms class,  use all plugins through hooks.
- `as_dict` function export data for activated plugins

Please review.

Reply:
lgtm. the plugin hook seems to be the right approach when designing the new version of api.